src/serializers/ng-snapshot.ts

```diff
@@ -0,0 +1,103 @@
+import type { Colors } from 'pretty-format';
+
+// jest hasn't exported this type yet so temporary we copy from 'pretty-format'
+interface PluginOptions {
+  edgeSpacing: string;
+  min: boolean;
+  spacing: string;
+}
+declare type Indent = (arg0: string) => string;
+declare type Print = (arg0: unknown) => string;
+
+const printAttributes = (
+  val: any, // eslint-disable-line @typescript-eslint/no-explicit-any
+  attributes: string[],
+  _print: Print,
+  indent: Indent,
+  colors: Colors,
+  opts: PluginOptions,
+): string => {
+  return attributes
+    .sort()
+    .map((attribute) => {
+      return (
+        opts.spacing +
+        indent(colors.prop.open + attribute + colors.prop.close + '=') +
+        colors.value.open +
+        (val.componentInstance[attribute] && val.componentInstance[attribute].constructor
+          ? `{[Function ${val.componentInstance[attribute].constructor.name}]}`
+          : `"${val.componentInstance[attribute]}"`) +
+        colors.value.close
+      );
+    })
+    .join('');
+};
+
+const ivyEnabled = (): boolean => {
+  // Should be required lazily, since it will throw an exception
+  // `Cannot resolve parameters...`.
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-var-requires
+  const { ɵivyEnabled } = require('@angular/core');
+
+  return !!ɵivyEnabled;
+};
+
+// Ivy component definition was stored on the `ngComponentDef`
+// property before `9.0.0-next.10`. Since `9.0.0-next.10` it was
+// renamed to `ɵcmp`.
+// eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-return
+const getComponentDef = (type: any): any => type.ngComponentDef || type.ɵcmp;
```

Comment:
Can we write basic return types for this?

Reply:
done